### PR TITLE
R-package cv

### PR DESCRIPTION
**Pull request by brsoyanvn**
Implemented catboost.cv function in R-package.

I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en.

### CODE REVIEW

File: catboost/R-package/src/catboostr.cpp [hunk 1]
```diff
@@ -339,6 +341,75 @@ SEXP CatBoostFit_R(SEXP learnPoolParam, SEXP testPoolParam, SEXP fitParamsAsJson
...
+        SET_STRING_ELT(columnNames, metricIdx * 4 + 0, mkChar(("test-" + metricName + "-mean").c_str()));
+        SET_STRING_ELT(columnNames, metricIdx * 4 + 1, mkChar(("test-" + metricName + "-std").c_str()));
+        SET_STRING_ELT(columnNames, metricIdx * 4 + 2, mkChar(("train-" + metricName + "-mean").c_str()));
+        SET_STRING_ELT(columnNames, metricIdx * 4 + 3, mkChar(("train-" + metricName + "-std").c_str()));
+    }
+
+    setAttrib(result, R_NamesSymbol, columnNames);
+    
+    R_API_END();
+    UNPROTECT(2);
```
Comment (nikitxskv) [inline catboost/R-package/src/catboostr.cpp]: It seems that you forgot to make UNPROTECT for variables in loop. You need to write so:
```
UNPROTECT(metricCount * 4 + 2);
```

**Comment by brsoyanvn** [comment]
@nikitxskv I fixed the issues you pointed me to.

**Comment by nikitxskv** [comment]
@arcadia-devtools Ship it!

**Comment by annaveronika** [comment]
We are currently planning our next release. For now the planned date is Friday. It would be super nice to have this pr in. Please do the changes asap.

**Comment by arcadia-devtools** [comment]
@nikitxskv, internal review request created: 645875

**Comment by annaveronika** [comment]
There is a build error 

$(SOURCE_ROOT)/catboost/R-package/src/catboostr.cpp:439:5: error: no matching function for call to 'CrossValidate'
    CrossValidate(
    ^~~~~~~~~~~~~
$(SOURCE_ROOT)/catboost/libs/train_lib/cross_validation.h:131:6: note: candidate function not viable: no known conversion from 'NCB::TDataProvider' (aka 'TDataProviderTemplate<NCB::TObjectsDataProvider>') to 'NCB::TDataProviderPtr' (aka 'TIntrusivePtr<TDataProviderTemplate<NCB::TObjectsDataProvider> >') for 4th argument
void CrossValidate(
     ^
1 error generated.

**Comment by annaveronika** [comment]
@brsoyanvn There is a build error, could you fix it? Do you see it when building?

**Comment by Evgueni-Petrov-aka-espetrov** [comment]
Hopefully, the error will go away if dereferencing is removed...

**Comment by brsoyanvn** [comment]
@annaveronika There is no any build error now

**Comment by annaveronika** [comment]
@brsoyanvn Do you plan to finish this?

**Comment by brsoyanvn** [comment]
@annaveronika sorry for a delay, I was busy with other issues. I resolved the conflicts

**Comment by annaveronika** [comment]
@arcadia-devtools Ship it!

**Pull request merged**